Blind the ciphertext properly

LLFourn opened the pull request:
Blind by a class group scalar rather than secp256k1 scalar.
See comment in hsm_cl.rs.

Unfortunately, this means not using an explicit rng but instead just KZen `sample_below` so there is an unused `rng` argument now. I didn't delete it because that would have to flow up several callers and it's fine to pass it in even if it's unused I think.

LLFourn committed:
Blind the ciphertext properly

Blind by a class group scalar rather than secp256k1 scalar.
See comment in hsm_cl.rs.

luckysori reviewed (approved):
LGTM. Thank you, Lloyd!

LLFourn committed:
Put class group discriminant size into const

Also what we are using should be referred to as a CL-group not a class
group. The CL-group is constructed within the class group.
